core/src/main/java/org/elasticsearch/index/analysis/NamedAnalyzer.java

```diff
@@ -104,4 +106,17 @@ public void setReusableComponents(Analyzer a, String f, TokenStreamComponents c)
             throw new IllegalStateException("NamedAnalyzer cannot be wrapped with a wrapper, only a delegator");
         }
     };
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof NamedAnalyzer)) return false;
+        NamedAnalyzer that = (NamedAnalyzer) o;
+        return Objects.equals(name, that.name);
+    }
```

Comment:
I don't have a better idea, but this makes me slightly nervous since nothing in the API prevents you from doing

```java
Analyzer a1 = new NamedAnalyzer("foo", someAnalyzer);
Analyzer a2 = new NamedAnalyzer("foo", otherAnalyzer);
```